### PR TITLE
[DataGrid] Escape formulas in CSV and Excel export

### PR DESCRIPTION
**Pull request by cherniavskii**
Closes https://github.com/mui/mui-x/issues/11702

- [x] docs

### CODE REVIEW

File: packages/x-data-grid/src/models/gridExport.ts [hunk 1]
```diff
@@ -40,6 +40,12 @@ export interface GridFileExportOptions<Api extends GridApiCommon = GridApiCommun
    * @returns {GridRowId[]} The list of row ids to export.
    */
   getRowsToExport?: (params: GridGetRowsToExportParams<Api>) => GridRowId[];
+  /**
```
Comment (flaviendelangle) [inline packages/x-data-grid/src/models/gridExport.ts:43]: We should probably document more why `false` is the default if it is recommended to set this to `true` :+1: 
Will the default become `true` in the next major?
If not, explain what is preventing it in the JSDoc.

If we don't explain our reasonning, it looks like we are not securing our library by default.

Comment (cherniavskii) [reply packages/x-data-grid/src/models/gridExport.ts:43]: Thanks for looking into it!
I'm leaning towards setting it to `true` by default, but I wanted to discuss this with the team first as it's technically a breaking change 👍🏻

File: packages/x-data-grid/src/models/gridExport.ts [hunk 2]
```diff
@@ -40,6 +40,12 @@ export interface GridFileExportOptions<Api extends GridApiCommon = GridApiCommun
...
+   * If `true`, the formulas in the cells will be escaped.
+   * It is recommended to set this to `true` for security reasons.
+   * @default false
```
Comment (cherniavskii) [inline packages/x-data-grid/src/models/gridExport.ts:46]: @mui/xgrid What do you think about making it `true` by default?
Technically, it's a breaking change, but it's also a security fix.
Any objections?

Comment (romgrk) [reply packages/x-data-grid/src/models/gridExport.ts:46]: No objection, we should change it.

File: packages/x-data-grid/src/models/gridExport.ts [hunk 3]
```diff
@@ -40,6 +40,12 @@ export interface GridFileExportOptions<Api extends GridApiCommon = GridApiCommun
...
+   */
+  escapeFormulae?: boolean;
```
Comment (romgrk) [inline packages/x-data-grid/src/models/gridExport.ts:48]: `escapeFormulas`? From what I can read online formulas seems to be the most popular plural for formula.

**Review by romgrk — APPROVED** [review]

File: packages/x-data-grid/src/models/gridExport.ts (hunk 3 above)
Comment (cherniavskii) [reply packages/x-data-grid/src/models/gridExport.ts:48]: I followed the same naming that `exceljs` used:

https://github.com/mui/mui-x/blob/760ea7e844b2d985d78a9a45295ca1b050defb11/packages/x-data-grid-premium/src/hooks/features/export/serializer/excelSerializer.ts#L123

But I actually prefer `escapeFormulas`, I'll update 👍🏻

**Review by michelengelen — APPROVED** [review]

**Pull request merged**